ocaml/xapi/helpers.ml

```diff
@@ -793,14 +793,8 @@ let max_version_in_pool : __context:Context.t -> int list =
   in
   List.fold_left max_version [] versions
 
-let rec string_of_int_list : int list -> string = function
-  | [] ->
-      ""
-  | x :: xs ->
-      if xs == [] then
-        string_of_int x
-      else
-        string_of_int x ^ "." ^ string_of_int_list xs
+let string_of_int_list =
+  List.fold_left (fun acc x -> Printf.sprintf "%s.%s" (string_of_int x) acc) ""
```

Comment:
I wonder if it's worth using
```
List.map string_of_int intlist |> String.concat "."
```

Reply:
I think that it's also reversing the list and appending a `.` at the end. I like Pau's suggestion.

Reply:
```